src/libraries/System.Text.Json/Common/JsonSimpleNamingPolicy.cs

```diff
@@ -0,0 +1,147 @@
+                ? ArrayPool<char>.Shared.Rent(bufferLength)
+                : null;
+
+            int resultLength = 0;
+            Span<char> result = buffer is null
+                ? stackalloc char[512]
+                : buffer;
+
+            void WriteWord(ref Span<char> result, ReadOnlySpan<char> word)
+            {
+                if (word.IsEmpty)
+                    return;
+
+                int required = result.IsEmpty
+                    ? word.Length
+                    : word.Length + 1;
+
+                if (required >= result.Length)
+                {
+                    int bufferLength = result.Length * 2;
+                    char[] bufferNew = ArrayPool<char>.Shared.Rent(bufferLength);
+
+                    result.CopyTo(bufferNew);
+
+                    if (buffer is not null)
+                        ArrayPool<char>.Shared.Return(buffer);
+
+                    buffer = bufferNew;
+                }
+
+                if (resultLength != 0)
+                {
+                    result[resultLength] = _boundary;
+                    resultLength += 1;
+                }
+
+                Span<char> destination = result.Slice(resultLength);
+
+                if (_lowercase)
+                {
+                    word.ToLowerInvariant(destination);
+                }
+                else
+                {
+                    word.ToUpperInvariant(destination);
+                }
+
+                resultLength += word.Length;
+            }
+
+            int first = 0;
+            ReadOnlySpan<char> chars = name.AsSpan();
+            CharCategory previousCategory = CharCategory.Boundary;
+
+            for (int index = 0; index < chars.Length; index++)
+            {
+                char current = chars[index];
+                UnicodeCategory currentCategoryUnicode = char.GetUnicodeCategory(current);
+
+                if (currentCategoryUnicode == UnicodeCategory.SpaceSeparator ||
+                    currentCategoryUnicode >= UnicodeCategory.ConnectorPunctuation &&
+                    currentCategoryUnicode <= UnicodeCategory.OtherPunctuation)
+                {
+                    WriteWord(ref result, chars.Slice(first, index - first));
+
+                    previousCategory = CharCategory.Boundary;
+                    first = index + 1;
+
+                    continue;
+                }
+
+                if (index + 1 < chars.Length)
+                {
+                    char next = chars[index + 1];
+                    CharCategory currentCategory = currentCategoryUnicode switch
+                    {
+                        UnicodeCategory.LowercaseLetter => CharCategory.Lowercase,
+                        UnicodeCategory.UppercaseLetter => CharCategory.Uppercase,
+                        _ => previousCategory
+                    };
+
+                    if (currentCategory == CharCategory.Lowercase && char.IsUpper(next) ||
+                        next == '_')
```

Comment:
Hello, Nino (: Thank you for the valuable review.

That's a trimmed version of the original implementation which used word boundary and uppercase-lowercase logic to get words from an input string, but it was too heavy for a name policy as @ericstj told me in the previous attempt or in the issue. Therefore, the current implementation only support CLR member names and not any input string.

Reply:
I don't recall the exact conversation but it may have been about bringing in new library dependencies to handle word boundaries?  Ultimately my guidance is usually just a suggestion and not a hard requirement -- we make the hard calls in API review and ultimately in PR with the area owners.  Not everything is settled and if there is new data that invalidates past decisions it can be revisited.  We do prioritize the right extensibility points in JSON rather than implementing every feature.  cc @dotnet/area-system-text-json